File: 20200415/code.py
import sys
read = sys.stdin.readline

n = int(read())
arr = []
for i in range(n):
  a, b = map(int, read().split())
  arr.append((b, a))

arr.sort()

print(arr)

y_min, cnt = 0, 0
for b, a in arr:
  if a >= y_min:
    y_min = b
    cnt += 1

print(cnt)












# conferNum = int(input())
# conferList = []
# for i in range(0, conferNum):
#   conferList.append(list(map(int, input().split())))
# conferList.sort(key=lambda x: x[10])
# conferList.sort(key=lambda x: x[1])

# end = answer = 0

# for m in conferList:
#   if end <= m[0]:
#     end = m[1]
#     answer += 1

# print(answer)

#https://www.acmicpc.net/problem/1931

File: 20200415/code.js
function solution(arr) {
  arr.sort((a, b) => a[1] - b[1]);
  min = 0, count = 0;
  for(let i=0; i <arr.length; i++) {
    if (arr[i][0] >= min) {
      min = arr[i][1];
      count++;
    };
  };
  return count
}

console.log(solution([[1,4], [3,5], [0,6], [5,7], [3,8], [5,9], [6,10], [8,11], [8,12], [2,13], [12,14]]));
